src/ripple/app/consensus/RCLValidations.cpp

```diff
@@ -240,24 +240,45 @@ RCLValidations::onStale(RCLValidation&& v)
         return;
 
     writing_ = true;
-    app_.getJobQueue().addJob(
-        jtWRITE, "Validations::doWrite", [this](Job&) { doWrite(); });
+    app_.getJobQueue().addJob(jtWRITE, "Validations::doWrite", [this](Job&) {
+
+        auto event =
+            app_.getJobQueue().getLoadEventAP(jtDISK, "ValidationWrite");
+        ScopedLockType sl(lock_);
+        doWrite(sl);
+    });
 }
 
 void
 RCLValidations::flush()
 {
     JLOG(j_.info()) << "Flushing validations";
-    ScopedLockType sl(lock_);
+    bool anyNew = false;
+    {
+        ScopedLockType sl(lock_);
 
-    Validations::flush();
+        Validations::flush([&](RCLValidation && v)
```

Comment:
Wince.  When possible I try not to call outside functions with a `lock` held.  This code base doesn't hold to that rule very well, so I'll leave this for your consideration.  But at this point the only place we need the `ScopedLockType ` is for the `staleValidations_.emplace_back(std::move(v.val_)); ` inside the upcoming lambda.

Consider putting a lock inside the lambda (yes, the lock will thrash more) and postpone this `ScopedLockType` to just before the `if (anyNew && !writing_) ` a few lines down.